### PR TITLE
fix(parser): improve canonical output and parser handling for edge cases

### PR DESCRIPTION
**Pull request by spencerbeggs**
## Summary

Raises raw yaml-test-suite compliance from 97.79% to 98.45% (19/1226 remaining canonical-output mismatches). Clears 9 fixtures from SKIP_ASSERTIONS: 5WE3, 6SLA, 6M2F, 9MMW, Q9WF, R4YG, T5N4, WZ62, X38W.

### Stringifier improvements

- Explicit `? key` / `: value` syntax now fires for scalar keys whose value contains a newline and for keys with `block-literal` / `block-folded` style. Previously only `YamlMap` and `YamlSeq` keys triggered this form.
- Block-style scalar keys (`|` / `>`) no longer get extra continuation-line indent; the renderer already bakes the indent into its rendered output.
- Block sequences and block mappings used as values under explicit `?` keys use compact notation: first item / first pair on the colon line, rest aligned via the configured indent.
- `renderBlockFolded` emits the explicit indent indicator (`>2`) when the value starts with two or more empty lines plus actual content (mirrors the existing `renderBlockLiteral` rule with a stricter threshold).

### Lexer / composer fixes

- Lexer recognises `:` adjacent to `}` or `]` in flow context as the value indicator (YAML 1.2 §7.18 flow-map adjacent value). Adds an `afterFlowCollectionEnd` flag mirroring the existing `afterQuotedScalar` flag.
- `flattenFlowChildren` flushes pending tag/anchor as an empty scalar at the `,` separator so metadata cannot bleed into the next flow-item.
- `flattenBlockMapChildren` flush-to-null heuristic now requires the trailing `:` to be on the same line as the scalar. The prior heuristic misfired on patterns like `? a\n: &b b\n: *a` where the second `:` belongs to a subsequent pair on a different line.

### Test post-processor

- `applySingleDocCanonical` re-renders a block-scalar root with `\n\t` content as a single-line double-quoted scalar with no `---`. Matches libyaml's conservative canonical form for tab-vs-indent ambiguity. Companion fixture M9B4 (same content, no `---` source) keeps block form, so the rule is specific to document-start position.

### Deferred

M5DY, KK5P, M2N8/00, M2N8/01 still fail canonical output. Investigated this session — the parser doesn't group explicit-`?` content as a CST subtree, so `? - <seq-key>\n: <value>` produces flat siblings instead of a structured pair. A focused parser refactor in `parseBlockMapping` is needed; deferred to a separate session because the existing composer leniency validations (which reject block-seq in key position for ZVH3) need coordinated updates.

## Test plan

- [x] `pnpm run test` (1149 unit + 1207 e2e passing)
- [x] `pnpm run typecheck`
- [x] `pnpm exec biome check`
- [x] Raw compliance: 1207/1226 passing (98.45%, up from 1199/1226 / 97.79% at branch start)
- [ ] CI pipeline green

Signed-off-by: C. Spencer Beggs <spencer@beggs.codes>

### CODE REVIEW

File: src/utils/lexer.ts [hunk 1]
```diff
@@ -92,6 +92,7 @@ export function createScanner(text: string): YamlScanner {
 	let afterEmptyBlockScalar = false;
 	/** Set when the previous token was a quoted scalar (single or double quoted). */
 	let afterQuotedScalar = false;
+	let afterFlowCollectionEnd = false;
```
[the remark this answers is not in the document]
Comment (spencerbeggs) [reply src/utils/lexer.ts:95]: ✅ Fixed in `b875e64394615ee76eef8dda661ae3ff26b65040`. Thanks!
